api/tests/unit/environments/identities/test_unit_identities_views.py

```diff
@@ -525,396 +525,439 @@ def test_identities_endpoint_returns_value_for_segment_if_identity_in_segment_an
+
+
+@override_settings(EDGE_API_URL="http://localhost")
+@mock.patch("environments.identities.views.forward_identity_request")
+def test_post_identities_calls_forward_identity_request_with_correct_arguments(
+    mocked_forward_identity_request: mock.MagicMock,
+    identity: Identity,
+    api_client: APIClient,
+    environment: Environment,
+    project: Project,
+) -> None:
+    # Given
+    url = reverse("api-v1:sdk-identities")
+
+    project.enable_dynamo_db = True
+    project.save()
+
+    data = {
+        "identifier": identity.identifier,
+        "traits": [
+            {"trait_key": "my_trait", "trait_value": 123},
+            {"trait_key": "my_other_trait", "trait_value": "a value"},
+        ],
+    }
+
+    # When
+    api_client.credentials(HTTP_X_ENVIRONMENT_KEY=environment.api_key)
+    api_client.post(url, data=json.dumps(data), content_type="application/json")
+
+    # Then
+    args, kwargs = mocked_forward_identity_request.delay.call_args_list[0]
+    assert args == ()
+    assert kwargs["args"][0] == "POST"
+    assert kwargs["args"][1].get("X-Environment-Key") == environment.api_key
+    assert kwargs["args"][2] == environment.project.id
+
+    assert kwargs["kwargs"]["request_data"] == data
+
+
+@override_settings(EDGE_API_URL="http://localhost")
+@mock.patch("environments.identities.views.forward_identity_request")
+def test_get_identities_calls_forward_identity_request_with_correct_arguments(
+    mocked_forward_identity_request: mock.MagicMock,
+    identity: Identity,
+    api_client: APIClient,
+    environment: Environment,
+    project: Project,
+) -> None:
+    # Given
+    project.enable_dynamo_db = True
+    project.save()
+
+    base_url = reverse("api-v1:sdk-identities")
+    url = base_url + "?identifier=" + identity.identifier
+
+    # When
+    api_client.credentials(HTTP_X_ENVIRONMENT_KEY=environment.api_key)
+    api_client.get(url)
+
+    # Then
+    args, kwargs = mocked_forward_identity_request.delay.call_args_list[0]
+    assert args == ()
+    assert kwargs["args"][0] == "GET"
+    assert kwargs["args"][1].get("X-Environment-Key") == environment.api_key
+    assert kwargs["args"][2] == project.id
+
+    assert kwargs["kwargs"]["query_params"] == {"identifier": identity.identifier}
+
+
+def test_post_identities_with_traits_fails_if_client_cannot_set_traits(
+    identity: Identity,
+    environment: Environment,
+    api_client: APIClient,
+    feature: Feature,
+) -> None:
+    # Given
+    url = reverse("api-v1:sdk-identities")
+    data = {
+        "identifier": identity.identifier,
+        "traits": [{"trait_key": "foo", "trait_value": "bar"}],
+    }
+
+    environment.allow_client_traits = False
+    environment.save()
+
+    # When
+    api_client.credentials(HTTP_X_ENVIRONMENT_KEY=environment.api_key)
+    response = api_client.post(
+        url, data=json.dumps(data), content_type="application/json"
+    )
+
+    # Then
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
```

Comment:
We should probably add a check to make sure the new traits don't exists

Reply:
Good idea! Added.